packages/react-components/react-progress/src/components/Progress/useProgress.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import { getNativeElementProps, resolveShorthand, useId } from '@fluentui/react-utilities';
+import type { ProgressProps, ProgressState } from './Progress.types';
+import { Label } from '@fluentui/react-label';
+import { progressCssVars } from './useProgressStyles';
+
+// If the percentComplete is near 0, don't animate it.
+// This prevents animations on reset to 0 scenarios.
+const ZERO_THRESHOLD = 0.01;
+
+/**
+ * Create the state required to render Progress.
+ *
+ * The returned state can be modified with hooks such as useProgressStyles_unstable,
+ * before being passed to renderProgress_unstable.
+ *
+ * @param props - props from this instance of Progress
+ * @param ref - reference to root HTMLElement of Progress
+ */
+export const useProgress_unstable = (props: ProgressProps, ref: React.Ref<HTMLElement>): ProgressState => {
+  // Props
+  const { barThickness = 'default', determinate = false, percentComplete } = props;
+  const ariaValueMin = determinate ? 0 : undefined;
+  const ariaValueMax = determinate ? 100 : undefined;
+  const ariaValueNow = determinate ? Math.floor(percentComplete! * 100) : undefined;
```

Comment:
Don't use the non-null assertion `!` for percentComplete. You should have `percentComplete` default to 0:
```suggestion
  const { barThickness = 'default', determinate = false, percentComplete = 0 } = props;
  const ariaValueMin = determinate ? 0 : undefined;
  const ariaValueMax = determinate ? 100 : undefined;
  const ariaValueNow = determinate ? Math.floor(percentComplete * 100) : undefined;
```